opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -1140,16 +1136,26 @@ namespace Opm {
         // For no well active globally we simply return.
         if( !wellsActive() ) return ;
 
+        const int reportStepIdx = ebosSimulator_.episodeIndex();
+        const Group& fieldGroup = schedule().getGroup("FIELD", reportStepIdx);
+
         // update group controls
         if (checkGroupControl) {
-            const int reportStepIdx = ebosSimulator_.episodeIndex();
-            const Group& fieldGroup = schedule().getGroup("FIELD", reportStepIdx);
             checkGroupConstraints(fieldGroup, deferred_logger);
         }
 
         for (const auto& well : well_container_) {
             well->updateWellControl(ebosSimulator_, well_state_, deferred_logger);
         }
+
+        // the group target reduction rates needs to be update since wells may have swicthed to/from GRUP control
+        // Currently the group targer reduction does not honor NUPCOL
+        if( localWellsActive() ) {
+            std::vector<double> groupTargetReduction(numPhases(), 0.0);
```

Comment:
I think I understand why you have moved this block out from the nupcol-related if: to ensure that we change the group target reduction when a well is moved in/out of group control. But keeping the targets fixed for wells under group control should be ok (under the assumption that nupcol is "large enough" to make the targets reasonable). If wells A and B get rates 900 and 800 at the nupcol deadline, then B switches to BHP and produces 780, it is still reasonable that A produces 900 and the acceleration nupcol is supposed to give requires that we keep the target fixed.